service/storage_service.cc

```diff
@@ -435,6 +435,8 @@ future<> storage_service::merge_topology_snapshot(raft_topology_snapshot snp) {
        return m.to_mutation(s);
    });
    co_await _db.local().apply(freeze(muts), db::no_timeout);
+   co_await topology_state_load(); // reload new state
+   _topology_state_machine.event.signal();
 }
```

Comment:
This is not needed. I is done in `group0_state_machine::load_snapshot`

Reply:
Right. I'm a bit worried though. There's this interval of time after `merge_topology_snapshot` but before Raft applies the snapshot descriptor. During this time, the mutations are already applied on-disk, but the in-memory representation of topology is not updated. We unlock the data after we pull the snapshot so observers might see this outdated in-memory representation (and if we crash and restart, we will start with the updated representation loaded from disk). But maybe it doesn't matter. For now I don't see any particular problem with this, at least regarding CDC generations.